src/components/policy/policy_regular/src/update_status_manager.cc

```diff
@@ -173,7 +178,11 @@ void UpdateStatusManager::DoTransition() {
   current_status_ = next_status_;
   next_status_.reset();
   LOG4CXX_DEBUG(logger_, "last_processed_event_ = " << last_processed_event_);
-  if (last_processed_event_ != kScheduleManualUpdate) {
+  const bool is_update_pending =
+      (policy::kUpdateNeeded == current_status_->get_status_string() &&
+       policy::StatusUpdatePending == current_status_->get_status());
```

Comment:
@mked-luxoft this thing looks really confusing. Could you create `kUpdatePending` state as you did for `PolicyTableStatus`?

Reply:
since name `StatusUpdatePending`, added `StatusProcessingSnapshot` in d35964dff07fd3af91d33ba543064bfddabb1446